internal/temporal/worker_deployment.go

```diff
@@ -26,6 +29,20 @@ type VersionInfo struct {
 	DrainedSince   *time.Time
 	TaskQueues     []temporaliov1alpha1.TaskQueue
 	TestWorkflows  []temporaliov1alpha1.WorkflowExecution
+
+	// True if all task queues in this version have at least one unversioned poller.
+	// False could just mean unknown / not checked / not checked successfully.
+	// Only checked for Target Version when Current Version is nil and strategy is Progressive.
+	// Used to decide whether to fast track the rollout; rollout will be AllAtOnce if:
+	//   - Current Version is nil
+	//   - Strategy is Progressive, and
+	//   - Presence of unversioned pollers in all task queues of target version cannot be confirmed.
+	AllTaskQueuesHaveUnversionedPoller bool
+	// True if all task queues in this version have no versioned pollers.
+	// False could just mean unknown / not checked / not checked successfully.
+	// Only checked for Drained versions that don't have controller-managed Deployments.
+	// Used to compute status.VersionCountIneligibleForDeletion.
+	NoTaskQueuesHaveVersionedPoller bool
```

Comment:
this is the meat of the new info that we're getting from server, what it's for, and when we get it